app/notifications/process_notifications.py

```diff
@@ -70,19 +70,24 @@ def persist_notification(template_id,
     return notification
 
 
-def send_notification_to_queue(notification, research_mode):
-    try:
-        research_mode = research_mode or notification.key_type == KEY_TYPE_TEST
+def send_notification_to_queue(notification, research_mode, queue=None):
+    if research_mode or notification.key_type == KEY_TYPE_TEST:
+        queue_to_use = 'research-mode'
+    elif queue is not None:
+        queue_to_use = queue
+    else:
```

Comment:
other minor thing, but i'd probably use `queue` rather than `queue_to_use` inside the function - then you can get rid of the `elif queue is not None:` block if you change this to `elif not queue:`